.github/workflows/pr-workflow.yaml

```diff
@@ -0,0 +1,42 @@
+# This workflow is called for every pull request
+# regardless which branch its targets
+
+name: Pull Request pipeline
+
+on:
+  pull_request
+
+jobs:
+  check-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      changes-maven: ${{ steps.check-changed-dirs.outputs.changes-maven }}
+      changes-coral: ${{ steps.check-changed-dirs.outputs.changes-coral }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
+        with:
+          ref: ${{ github.ref }}
+          fetch-depth: 0
+
+      - id: check-changed-dirs
+        name: Check directories of changes
+        run: |
+          if git diff --name-only HEAD^ HEAD | grep -q '^coral/'; then
+            echo "Changes detected in coral directory 🪸"
+            echo "changes-coral=true" >> $GITHUB_OUTPUT
+          fi
+          if git diff --name-only HEAD^ HEAD | grep -qv '^coral/'; then
+            echo "Changes detected for core/api"
```

Comment:
I believe the check here would more accurately translate to: "changes detected in somewhere else than coral directory."

For example, this would match if i change the root README.md and maven workflow would run.

What do you think, would it be better if we explicitly state the directories here as well? The relevant ones seem to be core and cluster-api, but maybe @aindriu-aiven has better understanding on the relevant files.

Reply:
Yes, that's true!

And I agree it would be good to have a more explicit check, that way we can avoid running unnecessary jobs.

As far as I see, I would also say the maven pipeline is needed for changes in `core` and `cluster-api`, but not really outside of it.

Reply:
btw today our merge with the release had a failed pipeline but still was comitted to main 🤔  this is something I want to read up tomorrow again. As far as I understood, having main as a protected branch and our workflow run in push to main, that should automatically prevent the commit if there's a failed pipeline, but maybe I understood that wrong ?( 

So I want to create a test-pipeline against a different branch and check that out (testing the "push on main" pipeline is kind of hard :D )